src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -1183,5 +1183,10 @@ void IInstaller.UpdateInstallMode(SdkFeatureBand sdkFeatureBand, bool? newMode)
             string newModeString = newMode == null ? "<null>" : newMode.Value ? WorkloadConfigCommandParser.UpdateMode_WorkloadSet : WorkloadConfigCommandParser.UpdateMode_Manifests;
             Reporter.WriteLine(string.Format(LocalizableStrings.UpdatedWorkloadMode, newModeString));
         }
+
+        // This method should never be called for this kind of installer. It is challenging to get this information from an MSI
+        // and totally unnecessary as the information is identical from a file-based installer. It was added to IInstaller only
+        // to facilitate testing. As a consequence, it does not need to be implemented.
+        public WorkloadSet GetWorkloadSetContents(string workloadVersion) => throw new NotImplementedException();
```

Comment:
We actually do this already for MSIs.  Check out the `ExtractManifestAsync` implementation.  In fact, you may be able to implement `GetWorkloadSetContents` in terms of that, possibly only in test code instead of in the product.

Reply:
I don't know if I'm understanding this correctly, but it looks like it extracts information by installing the MSI?

https://github.com/dotnet/sdk/blob/0113afe0e7777a01a21cc7a985fad9075734a660/src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs#L792

That seems like it would change what you have installed, which wouldn't necessarily be what we'd want.

Reply:
It does an "admin install", which means it doesn't actually install it, it just extracts the MSI to the target path.